lib/modules/navigationMiddleware.js

```diff
@@ -2,45 +2,55 @@ import pathToRegex from 'path-to-regexp';
 import { METHODS } from './router';
 import * as actions from './actions';
 
+export const parseRoute = (url, routes) => {
+  let route;
+
+  for (route of routes) {
+    const [url, handler] = route;
+    const reg = pathToRegex(url);
+    const result = reg.exec(url);
+
+    if (result) {
+      return { handler, reg, result };
+    }
+  }
```

Comment:
No, I think it should return undefined if there was no match. Otherwise,
if(parseUrl(...)) may unexpectedly pass, as it's not false.
On May 31, 2016 5:42 PM, "Niranjan Ramadas" <notifications@github.com>
wrote:

> In lib/modules/navigationMiddleware.js
> <https://github.com/reddit/node-platform/pull/10#discussion_r65286913>:
>
> > @@ -2,45 +2,55 @@ import pathToRegex from 'path-to-regexp';
> >  import { METHODS } from './router';
> >  import * as actions from './actions';
> >
> > +export const parseRoute = (url, routes) => {
> > +  let route;
> > +
> > +  for (route of routes) {
> > +    const [url, handler] = route;
> > +    const reg = pathToRegex(url);
> > +    const result = reg.exec(url);
> > +
> > +    if (result) {
> > +      return { handler, reg, result };
> > +    }
> > +  }
>
> should we return an empty object if the for loops doesn't return?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/reddit/node-platform/pull/10/files/de3bd70dc07c433d2b5fb1e3611e55d27bf8e054#r65286913>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAKtm-_EaQkBdPZQ7VCXIIuvCzLLXatfks5qHNV9gaJpZM4IrFjP>
> .
>